### PR TITLE
HackPass, Checkin component

### PR DESCRIPTION
**Pull request by pierreTklein**
## Features

- Staff can check in users via QR code, or email ( #305 )
- Hackers can download their HackPass from the website, and also view their QR code. ( #22 ) 
- Generate HackPass pdf when user is checked in, which is formatted for sticker printout.

## Staff Dashboard

image

## Hacker Dashboard

image

## HackPass on website

image

## HackPass in PDF form

image

## Checkin Page

image

### CODE REVIEW

**Comment by erickzhao** [comment]
I'll try running through a full flow tonight.

File: src/Checkin/Email.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      initialValues={{ email: '' }}
+      onSubmit={handleSubmitFactory(props)}
+      render={renderFormik}
+      validationSchema={getValidationSchema()}
+    />
+  );
+};
+
+function handleSubmitFactory(
+  props: IEmailProps
```
Comment (erickzhao) [inline src/Checkin/Email.tsx]: Why pass in the entire props object instead of just the email?

File: src/api/hacker.ts [hunk 2]
```diff
@@ -100,11 +117,12 @@ class HackerAPI {
    * Update's a hacker's status to checked-in if the hacker is accepted or confirmed.
    * @param {String} id The id of the hacker to be updated
    */
-  public checkinStatus(id: string): AxiosPromise {
+  public async checkin(id: string): Promise<AxiosResponse<APIResponse<{}>>> {
     const key = CACHE_HACKER_KEY + '-' + id;
-    const value = API.getEndpoint(APIRoute.HACKER_CHECKIN).patch(
+    console.log(id);
```
Comment (erickzhao) [inline src/api/hacker.ts]: Is this console log necessary?

**Comment by pierreTklein** [comment]
@krubenok 
image
image

**Comment by krubenok** [comment]
@hackmcgill/design can we get a typeface & weight review on this. Layout looks good though!

**Pull request merged**